src/prymitive2.ts

```diff
@@ -0,0 +1,210 @@
+    _levels: Level [] = [];
+    _name: string;
+  
+    constructor(name:string, levels: Level []) {
+      this._name = name;
+      this._levels = levels
+    }
+
+    get levels () : Level [] {
+        return this._levels
+    }
+
+    get name () : string {
+        return this._name
+    }
+
+    addLevel (level:Level) : void {
+        this._levels.push(level)
+    }
+
+    removeLecturer (levelToRemove:Level) : void {
+        this._levels = this._levels.filter(item => item !== levelToRemove);
+        }  
+  }
+  
+  class Level {
+    // implement getters for fields and 'add/remove group' methods
+    _groups: Group[];
+    _name : string;
+    _description : string;
+  
+    constructor(name:string, description:string, groups:Group[]) {
+      this._name = name;
+      this._description = description;
+      this._groups = groups
+    }
+
+    get groups () : Group[] {
+        return this._groups
+    }
+
+    get name () : string {
+        return this._name
+    }
+
+    get description () : string {
+        return this._description
+    }
+
+    removeGroup (groupToRemove:Group) : void {
+        this._groups = this._groups.filter(group => group !== groupToRemove);
+        }   
+     
+    addGroup (groupToAdd:Group): void {
+        this._groups.push(groupToAdd)
+    }
+    }
+
+  
+  
+  class Group {
+
+    _area: string;
+    _status: ('active' | 'pending');
+    _students: Student[] = []; // Modify the array so that it has a valid toSorted method*
+    _directionName: string;
+    _levelName: string;
+  
+    constructor(directionName: string, levelName: string, area: string, status: ('active' | 'pending'), students: Student[]) {
+      this._directionName = directionName;
+      this._levelName = levelName;
+      this._area = area;
+      this._status = status;
+      this._students = students;
+    }
+
+    set status (groupStatus:('active' | 'pending')) {
+        this._status = groupStatus
+    }
+    get area () : string {
+        return this._area
+    }
+    get status () : string {
+        return this._status
+    }    
+    get studentst () : Student[] {
+        return this._students
+    }    
+    get directionName () : string {
+        return this._directionName
+    }    
+    get levelName () : string {
+        return this._levelName
+    }
+    addStudent(student: Student) : void {
+        this._students.push(student)
+    }
+    removeGroup (studentToRemove: Student) : void {
+        this._students = this._students.filter(item => item !== studentToRemove );
+    }    
+
+    showPerformance() :Student [] {
+      const sortedStudents = this._students.toSorted((a, b) => b.getPerformanceRating() - a.getPerformanceRating());
+      return sortedStudents;
+    }
+}
+  
+type grade = {[workName:string]:number}; //it can be not only 1 key-value up here, how to prevent that???
+type visit = {[lesson: string]:boolean};//it can be not only 1 key-value up here, how to prevent that???
```

Comment:
It depends on what you want to achieve. Try just:
`type visit = { lesson: boolean };`